x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/persistence/IndexBasedTransformConfigManager.java

```diff
@@ -327,6 +329,7 @@ public void getTransformConfigurationForUpdate(
             // use sort to get the last
             .addSort("_index", SortOrder.DESC)
             .setSize(1)
+            .setAllowPartialSearchResults(false)
```

Comment:
What does it mean "partial search results"? Results from a subset of nodes?

Reply:
It means that it will not throw if only a subset of the shards fail the search.

Reply:
subset of shards, which might be on different nodes, yes. In the default setting ES returns with 200 but actually might not have all results. This is somewhat "search engine behavior", if you care about consistency you can either ask for `getFailedShards()` in the response object or you disallow partial results which will make it throw. I prefer the more explicit behavior.

Note that per default the number of shards is 1 for this config index, therefore it should not make a difference in this case, because it should throw if all shards failed. But I think it's a good setting as the default might change.

Reply:
Ok, thanks for explanations.